### PR TITLE
[#156753589] Add tasks to the billable events

### PR DESCRIPTION
**Pull request by paroxp**
## What

Currently, the tasks are being ignored when it comes to converting data
into our defined standard. Tasks too cost money, therefore we'd like to
make sure those are standardised and counted towards any billing
calculations.

## How to review

- Run tests
- Run application against your environment
- Create and kill a task
- Request and see data (tested myself)

### CODE REVIEW

File: eventstore/sql/create_events.sql [hunk 1]
```diff
@@ -66,6 +66,30 @@ INSERT INTO events with
...
+				id as event_sequence,
+				guid::uuid as event_guid,
+				created_at,
+				(raw_message->>'task_guid')::uuid as resource_guid,
+				(raw_message->>'task_name') as resource_name,
+				'task'::text as resource_type,                              -- resource_type for compute resources
+				(raw_message->>'org_guid')::uuid as org_guid,
+				(raw_message->>'space_guid')::uuid as space_guid,
+				'ebfa9453-ef66-450c-8c37-d53dfd931038'::uuid as plan_guid,  -- plan guid for all compute resources
+				'task'::text as plan_name,                                  -- plan name for all compute resources
```
Comment (dcarley) [inline eventstore/sql/create_events.sql]: Should these comments say `for all task resources`?

File: eventstore/store_billable_events_test.go [hunk 2]
```diff
@@ -110,6 +110,95 @@ var _ = Describe("GetBillableEvents", func() {
...
+	 .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .
+	*-----------------------------------------------------------------------------------*/
+	It("Should return one BillingEvent for a task that was running for 1hr", func() {
+		cfg.AddPlan(eventstore.PricingPlan{
+			PlanGUID:  eventstore.TaskPlanGUID,
+			ValidFrom: "2001-01-01",
+			Name:      "PLAN1",
+			Components: []eventstore.PricingPlanComponent{
+				{
+					Name:         "compute",
```
[the remark this answers is not in the document]
Comment (paroxp) [reply eventstore/store_billable_events_test.go]: Yup. Shame the tests didn't blow due to that... Will change.

File: eventstore/sql/create_events.sql (hunk 1 above)
Comment (paroxp) [reply eventstore/sql/create_events.sql]: Good catch :D

**Comment by paroxp** [comment]
I've addressed the comments.

Regarding the additional tests, I too felt it would be unnecessary to duplicate the tests. I realise there could be edge cases.

I also didn't think a task would run for longer enough to start before and stop after the requested time period.

Not sure what the solution is. I'm happy to duplicate the other tests.

**Comment by dcarley** [comment]
> I also didn't think a task would run for longer enough to start before and stop after the requested time period.

Yeah, good point, because we're reporting on months. Starting before or ending after would be quite common though.

Maybe we should talk to @chrisfarms since he recently refactored all of the tests and one possible solution to this would be to have more magic/meta tests that check the same behaviours of both apps and tasks?

File: eventstore/store_billable_events_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcarley) [reply eventstore/store_billable_events_test.go]: They have now 😆 (Travis build failed because the name appears in the assertion too)

**Comment by dcarley** [comment]
I spoke to Chris. Something that I hadn't quite grokked before is that we're normalising the task states to `STARTED` and `STOPPED` which are meanings that are interpreted later in the processing when you might be missing one of those states. So it's probably fine to only test this scenario for tasks 👍 

Thanks for the fixes, merging 👍 

Do we need to release in paas-cf?
